src/NzbDrone.Core/MediaFiles/DiskScanService.cs

```diff
@@ -67,26 +67,29 @@ public void Scan(Series series)
         {
             var rootFolder = _diskProvider.GetParentFolder(series.Path);
 
-            if (!_diskProvider.FolderExists(rootFolder))
+            var rootFolderExists = _diskProvider.FolderExists(rootFolder);
+            if (!rootFolderExists)
             {
                 _logger.Warn("Series' root folder ({0}) doesn't exist.", rootFolder);
-                _eventAggregator.PublishEvent(new SeriesScanSkippedEvent(series, SeriesScanSkippedReason.RootFolderDoesNotExist));
+                _eventAggregator.PublishEvent(new SeriesScanSkippedEvent(series,
+                    SeriesScanSkippedReason.RootFolderDoesNotExist));
                 return;
             }
 
-            if (_diskProvider.GetDirectories(rootFolder).Empty())
+            if (_diskProvider.GetDirectories(rootFolder).Empty() && !_configService.UpdateSeriesIfFolderMissing && !_configService.CreateEmptySeriesFolders)
             {
-                _logger.Warn("Series' root folder ({0}) is empty.", rootFolder);
-                _eventAggregator.PublishEvent(new SeriesScanSkippedEvent(series, SeriesScanSkippedReason.RootFolderIsEmpty));
-                return; 
+                _logger.Warn("Series' root folder ({0}) is empty and no options set to create folders or clean DB.", rootFolder);
+                _eventAggregator.PublishEvent(new SeriesScanSkippedEvent(series,
+                    SeriesScanSkippedReason.RootFolderIsEmpty));
+                return;
             }
 
             _logger.ProgressInfo("Scanning disk for {0}", series.Title);
-            
-            if (!_diskProvider.FolderExists(series.Path))
+
+            var seriesFolderAlreadyExists = _diskProvider.FolderExists(series.Path);
```

Comment:
I did it this way so that the `_mediaFileTableCleanupService.Clean` call can still be done on the created folder so it's up to date, but I could do the check at that point and then exit early.